---
title: "soyaGWAS_Tutorial"
author: "Avjinder Kaler, Jaspreet Sandhu, Jugpreet Singh"
date: "10/04/2020"
output: html_document
---

Introduction:
This package can be used to subset the Hapmap formatted genotypic data for specific set of genotypes from USDA soybean diverse germplasm (50K SNP-chip data for 20K genotypes). For example, if you have phenotype data collected for 400 genotypes you can use this function to subset the genotypic SNP data and use it for downstream GWAS pipeline. The subset genotypic data is then used to run GWAS. The subsetting and GWAS can be performed using a single function `SGS.GWASss`. 

The package contain function to calculate allelic effects and find favorable alleles for a trait of interest in given subset. Further, another function in package can be used to mine favorable alleles and calculate the cumulative allelic effects for the entire USDA soybean diverse germplasm (50K SNP-chip data for 20K genotypes) collection.   


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("multtest")
install.packages("gplots")
install.packages("trainbr")
install.packages("genetics")
install.packages("LDheatmap")
install.packages("EMMREML")
install.packages("scatterplot3d")
install.packages("bigmemory")
install.packages("biganalytics")

library(multtest)
library(gplots)
library(LDheatmap)
library(genetics)
library(EMMREML)
library(compiler)
library("scatterplot3d")
library("bigmemory")
library("biganalytics")
source("http://zzlab.net/GAPIT/gapit_functions.txt")
source("http://zzlab.net/GAPIT/emma.txt")
source("http://zzlab.net/FarmCPU/FarmCPU_functions.txt")

library(data.table)
library(tidyr)
library(plyr)
library(dplyr)

```

Set up the working directory 
```{r}
setwd("SGS")
```

```{r, include=F}
myG=read.table("https://uark.box.com/shared/static/huxfqiunluv9k29its510bgqtsbqaxli.txt", header = FALSE)
dim(myG)
```


```{r}
myY <- read.table("Traitexample.txt", head = TRUE) 
dim(myY)
```


Subsetting myG
```{r}
myGss=SGS.Gss(G=myG,pheno=myY)
dim(myGss)
myGss[1:10,1:10]
```

Subsetting myG and performing GWAS

```{r}
myGWAS.out=SGS.GWASss(G=myG,pheno=myY,maf=0.01,PCA.total=3,threshold=1,model="FarmCPU")
myGWAS=myGWAS.out$GWAS

```


Allelic effect 
```{r}
myGWAS=read.csv("GAPIT.FarmCPU.GL.GWAS.Results.csv", header =TRUE )
myGss=read.table("Genosubset.hmp.txt", head = FALSE) # Geno file from A ; NEED THIS FORMAT FOR  GAPIT

mySGS.out=SGS.ale.eff(GWAS = myGWAS,Gss = myGss,pheno = myY,p = 0.0005)
myAllelic.eff=mySGS.out$allele.effect

```

Cumm Allelic effect 
```{r}
myGWAS=read.csv("GAPIT.FarmCPU.GL.GWAS.Results.csv", header =TRUE )
myAllelic.eff=read.delim("SGS.Trait.AllelicEffect.txt", head = TRUE) # Geno file from A ; NEED THIS FORMAT FOR  GAPIT

myEGS.cum.effa=SGS.cumeff(G = myGss,GWAS = myGWAS,allele.effect = myAllelic.eff,p = 0.005,trait.val = 1)

my.cum.effa=myEGS.cum.effa$Geno.cum.eff

```


